### PR TITLE
[Minor] Added Select all support to measurement client

### PR DESCRIPTION
**Pull request by MounikaBattu17**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md). _(Required)_
- [ ]  Automatically post PR comments with images for G code changes? _(Recommended for small changes)_

### What does this Pull Request accomplish?
Added Select All measurements support to Measurement Client tool.

### Why should this Pull Request be merged?
- Added a `Select All` checkbox in `Available Measurements` listbox header. Clicking on this icon should select all measurements in the list box.
- **Behaviour of this checkbox:**
    - `Select All` checkbox is disabled if no measurements are available.
    - Upon checking this checkbox, all measurements in the list are selected.
    - Upon unchecking this checkbox, all measurements in the list are unselected.
    - `Select All` checkbox is checked if all measurements in the list are selected.
    - `Select All` checkbox is unchecked if any measurement in the list is not selected.
    - Upon Refresh, `Select All` is either unchecked or, disabled if no measurements are available.
    - Other functionalities like client name validation, Create client button enable/disable, Warning visibility, will be working as expected for `Select All` changes.
- **Code changes:**
  - Added a new event for select All value change, which will update the measurements icons, client name text color, create client button state, warning visibility based on select all boolean.
  - Updated code in `Available Measurements Mouse Down` event to check/uncheck Select All checkbox based on measurements selection.
  - Updated `Refresh Value change` event to either disable(if no measurements available) or uncheck select All checkbox.

### What testing has been done?
Manually verified the Select All functionality.

### CODE REVIEW

**Comment by MounikaBattu17** [comment]
Select All support

**Comment by dixonjoel** [comment]
When I see the error message with the invalid characters, it doesn't seem to include '.' or '_'. Why is 'ni.examples.GameOfLife_Python' invalid?

**Comment by Avinash2Suresh** [comment]
> When I see the error message with the invalid characters, it doesn't seem to include '.' or '_'. Why is 'ni.examples.GameOfLife_Python' invalid?

The client name for 'ni.examples.GameOfLife_Python' is not unique. The active project already has an existing library with the same name.

**Comment by dixonjoel** [comment]
> > When I see the error message with the invalid characters, it doesn't seem to include '.' or '_'. Why is 'ni.examples.GameOfLife_Python' invalid?
> 
> The client name for 'ni.examples.GameOfLife_Python' is not unique. The active project already has an existing library with the same name.

That's a little misleading - the error message says it has invalid characters. Do we have a different message for non-unique library?

**Review by dixonjoel — APPROVED** [review]

**Review by Prem-MysteryPro — APPROVED** [review]

File: Source/Generator/Create Measurement Client.vi
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: Thanks for the suggestion. Updated.

**Comment by MounikaBattu17** [comment]
> > > When I see the error message with the invalid characters, it doesn't seem to include '.' or '_'. Why is 'ni.examples.GameOfLife_Python' invalid?
> > 
> > 
> > The client name for 'ni.examples.GameOfLife_Python' is not unique. The active project already has an existing library with the same name.
> 
> That's a little misleading - the error message says it has invalid characters. Do we have a different message for non-unique library?

Update the error message to provide better clarity of the error.

**Pull request merged**